sdk/openai/Azure.AI.OpenAI/src/Helpers/MultipartFormDataRequestContent.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.OpenAI
+{
+    internal sealed class MultipartFormDataRequestContent : RequestContent
+    {
+        public string Boundary { get; }
+        private MultipartFormDataContent _baseContent { get; }
+
+        public MultipartFormDataRequestContent() : base()
+        {
+            // Open: does this need to be random?
+            Boundary = "1234-multipart-boundary-token";
```

Comment:
I was wondering the same. If it isn't random, could it become a vector of attacks?

Reply:
As far as I can tell from the relevant RFC documents, the only constraint is that the boundary shouldn't ever be expected to naturally occur within the body of a request.

https://www.rfc-editor.org/rfc/rfc2046#section-5.1
https://www.rfc-editor.org/rfc/rfc7578#section-4.1

Randomization is a recommendation (particularly for generalized implementations) and I don't think having it hard-coded introduces any security concerns; any potential attack vectors with injected boundaries would already be possible with late-binding payload manipulation (a la custom pipeline policies) and the service should handle multipart message fragments accordingly.